### PR TITLE
5601 - fix next.js builds on node v16

### PR DESCRIPTION
**Pull request by luu-alex**
## Description
#5601 
this PR fixes builds on next.js when being built with node v16, I believe `global.request` was being [changed ](https://github.com/web3/web3.js/blob/1.x/packages/web3-providers-http/src/index.js#L31-L33) due to a polyfill issue and was giving an error `TypeError: Cannot set property Request of # which has only a getter` and its due to `require('abortcontroller-polyfill/dist/polyfill-patch-fetch')`

The solution is to only include `require('abortcontroller-polyfill/dist/polyfill-patch-fetch')` when abortController is not defined globally. I believe next.js provides their own global variables and would run into an error when `Request` was being polyfilled.

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../blob/1.x/CONTRIBUTIONS.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist for 1.x:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: packages/web3-providers-http/src/index.js [hunk 1]
```diff
@@ -27,10 +27,15 @@ var errors = require('web3-core-helpers').errors;
...
 
+// check if node isn't being run
+if (!typeof window === undefined) {
 // Apply missing polyfill for IE
 require('cross-fetch/polyfill');
 require('es6-promise').polyfill();
 require('abortcontroller-polyfill/dist/polyfill-patch-fetch');
+} else {
+    const fetch = require('cross-fetch');
+}
```
Comment (CodyJasonBennett) [inline packages/web3-providers-http/src/index.js]: Does this work the way you want it to? `fetch` is later accessed from global scope.

Comment (luu-alex) [reply packages/web3-providers-http/src/index.js]: Your right, fetch was still being undefined. removed const, thanks

**Review by avkos — CHANGES_REQUESTED** [review]
Some of the tests are falling. It will be great to fix them

**Comment by luu-alex** [comment]
@avkos  fixed a majority of failing testcases, created an issue for the tests still failing #5639

**Review by nikoulai — APPROVED** [review]

File: test/httpprovider.js [hunk 2]
```diff
@@ -130,7 +132,7 @@ describe('web3-providers-http', function () {
                     method: 'POST',
                     credentials: 'omit',
                     headers: {
-                      'Content-Type': 'application/json'
+                    'Content-Type': 'application/json'
```
Comment (Muhammad-Altabba) [inline test/httpprovider.js]: I think deleting those whitespaces happened by mistake :) 
```suggestion
                      'Content-Type': 'application/json'
```

**Review by Muhammad-Altabba — APPROVED** [review]

**Review by avkos — APPROVED** [review]

**Pull request merged**